applications/machine_learning/README.rst

```diff
@@ -0,0 +1,426 @@
+.. _nrf_machine_learning_app:
+
+nRF machine learning
+####################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The nRF machine learning application is an out of the box reference design of an :ref:`embedded machine learning` using :ref:`Edge Impulse`.
+The application performs the following tasks:
+
+* Gathers data from sensors
+* Forwards data to Edge Impulse
+* Runs machine learning model and displays results on LEDs
+
+Overview
+********
+
+As it was mentioned before, the nRF machine learning application performs multiple tasks.
+The tasks are described by the paragraphs below.
+
+Sampling sensors
+================
+
+Sensor sampling is handled by CAF Sensor sampler module.
+The mentioned CAF module uses :ref:`zephyr:sensor_api` to handle sensor sampling.
+This approach makes it easy to use any sensor available in Zephyr.
+
+By default, the following sensors are used by the application:
+
+* On **Thingy:52** - the built-in accelerometer (``LIS2DH``).
+* On **nRF52840 Development Kit**  - the simulated sensor (``sensor_sim`` available in |NCS|).
+  The simulated sensor generates predefined waves as acceleration.
+  This board does not have a built-in accelerometer.
+
+Forwarding data
+===============
+
+The application uses :ref:`Edge Impulse's data forwarder` protocol to forward data to :ref:`Edge Impulse studio` over :ref:`zephyr:uart_api` or :ref:`nus_service_readme`.
+
+By default, the following transports are used:
+
+* **Thingy:52** uses NUS.
+* **nRF52840 Development Kit** uses UART.
+
+Machine learning model
+======================
+
+The machine learning model is handled using :ref:`ei_wrapper` available in |NCS|.
+
+By default, the application uses pretrained models deployed in :ref:`Edge Impulse studio`:
+
+* On **Thingy:52** - the model uses built-in accelerometer data to recognize the following gestures:
+
+  * ``idle`` - the device is placed on a flat surface.
+  * ``updown`` - the device is moved in updown direction.
+  * ``rotate`` - the device is rotated.
+  * ``tap`` - the device is tapped while placed on a flat surface.
```

Comment:
Why are the gesture names in code formatting? I'd add an explanation where I can find these code names in the app/EI app.

Reply:
These names are specific to used machine learning model. In Edge Impulse studio you define labels that should be assigned to given gestures. I will add information about this.
In application source code this names are used by the `ml_result_event` and `led_state_def.h` file. Still, the application only refers to labels (strings) that are defined by the machine learning model.